packages/pages/src/util/processEnvVariables.ts

```diff
@@ -21,10 +21,43 @@ export const processEnvVariables = (
 
   return Object.fromEntries(
     Object.entries(loadEnv(mode, process.cwd(), prefix))
-      // For some reason this env var is automatically set and causes issues so
+      // For some reason this env var is automatically set and causes issues in Linux so
       // we filter it out specifically.
       .filter(([env]) => env !== "_")
+      // Filter out function keys as they cause problems with esbuild. For example, the
+      // key `'BASH_FUNC_protosearch%%'`, which is an exported function, results in the error:
+      // The define key "BASH_FUNC_protosearch%%" must be a valid identifier
+      .filter(([env]) => isIdentifier(env))
       // The value must be stringified: https://vitejs.dev/config/shared-options.html#define
       .map(([key, value]) => [key, JSON.stringify(value)])
   );
 };
+
+const startRunes = "_$abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+const continueRunes =
+  "_$0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
```

Comment:
will these change? if not, enums?

Reply:
They won't change. It's easier to have a list of strings to do an includes check than it is to define all as enums and switch case on them.